### PR TITLE
WIP: Auto-fixer for pyproject.toml

### PR DESCRIPTION
**Pull request by tiran**
- add missing pyproject.toml
- add or update [build-system] map
- detect additional build requirements by parsing "setup.py"
- remove unwanted build requirements such as cmake

Fixes: #260
Fixes: #274

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -11,5 +11,6 @@ resolvelib
 setuptools<72.0.0
 stevedore
 toml
+tomlkit
```
Comment (dhellmann) [inline requirements.txt:14]: Do we need both toml libraries? We don't have to remove the other one in this PR, but maybe a ticket to do that cleanup?

File: src/fromager/pyproject.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        self.req = req
+        self.root = sdist_root_dir
+        self.pyproject_toml = self.root / "pyproject.toml"
+        self.setup_py = self.root / "setup.py"
+
+    def update(self):
+        doc = self.load()
+        build_system = self.default_builds_system(doc)
+        self.update_build_requires(build_system)
+        self.save(doc)
```
Comment (dhellmann) [inline src/fromager/pyproject.py:64]: Does tomlkit preserve comments? Maybe we don't care if we roundtrip the file and lose info?

**Review by dhellmann — COMMENTED** [review]
Would you apply this to everything being built? Or is it meant to be an optional step only used for some packages?

File: src/fromager/pyproject.py (hunk 2 above)
Comment (tiran) [reply src/fromager/pyproject.py:64]: Yes, tomlkit preserves comments and even layout of the file. Fromager's `pyproject.toml` does roundtrip perfectly.

```python
>>> import pathlib, tomlkit, toml
>>> raw = pathlib.Path("pyproject.toml").read_text()
>>> tomlkit.dumps(tomlkit.loads(raw)) == raw
>>> print(tomlkit.dumps(tomlkit.loads(raw)))
>>> print(toml.dumps(toml.loads(raw)))
```
tomlkit keeps
```toml
[tool.setuptools.dynamic]
optional-dependencies.build = { file = ["requirements-build.txt"] }
optional-dependencies.mypy = { file = ["requirements-mypy.txt"] }
optional-dependencies.test = { file = ["requirements-test.txt"] }
dependencies = { file = ["requirements.txt"] }
```

while toml changes the lines to
```toml
[tool.setuptools.dynamic.optional-dependencies.build]
file = [ "requirements-build.txt",]

[tool.setuptools.dynamic.optional-dependencies.mypy]
file = [ "requirements-mypy.txt",]

[tool.setuptools.dynamic.optional-dependencies.test]
file = [ "requirements-test.txt",]
```

Comment (dhellmann) [reply src/fromager/pyproject.py:64]: Nice, let's switch!

Comment (tiran) [reply src/fromager/pyproject.py:64]: Let me make a separate PR for that.

**Comment by tiran** [comment]
Replaced by #407

**Pull request closed without merging**